core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -204,7 +225,7 @@ private void connectToChunkList(List<Chunk<K, V>> engaged, List<Chunk<K, V>> chi
                 break;
             }
             // chunk is not in list (someone else already updated list), so we're done with this part
-            if ((curr == null) || (prev == null)) {
+            if (curr == null) {
```

Comment:
if `prev` was initialized with null (line 209) we would search from head (lines 212-220). At this point, either `curr` is `firstEngaged` and `prev` is null (because the while loop in line 216 never happened), and the if clause in line 223 holds, or curr is necessarily not null.

Also, what @yoavz1997 said: this is an improvement suggested by IntelliJ static analyzer.